data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,241 @@
+        pid=`ps aux | grep "$(basename $0) -s" | grep -v "grep"| awk '{print $2}'`
+        if [[ -n $pid ]];then
+            #echo $pid
+            kill -9 $pid >> /dev/null 2>&1;sync
+        else
+            echo "PIDs was killed"
+        fi        
+        
+        echo "Kill all associated "virsh cnsole" processes"
+        ps aux | grep "virsh console" | grep -v "grep" >> /dev/null 2>&1
+        while [[ $? -eq 0 ]];    
+        do
+            sleep 10
+            sub_pid=`ps aux | grep "expect -c" | grep -v "grep"| awk '{print $2}'`
+            if [[ -n $sub_pid ]];then 
+                kill -9 $sub_pid >> /dev/null 2>&1;sync
+            else
+                break
+            fi    
+        done        
+}
+
+
+
+function end_monitor()
+{
+        #end of monitor for guest console
+		if [ -f ${MONITOR_LOCKED_FILE} ];then
+            echo "End of Monitor for guest console"	
+            #UNLOCK Monitor for guest console	    
+            > ${MONITOR_LOCKED_FILE} ; rm -rf ${MONITOR_LOCKED_FILE}
+            #Kill Pids from Monitor for guest console
+            close_console
+        fi
+}
+
+function start_monitor()
+{
+        #start monitor for guest console
+        get_vm_guestnames_state_running=`virsh list --state-running | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+        vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})  
+
+        while :
+        do
+		    if [ -f ${MONITOR_LOCKED_FILE} ];then
+                #Check with LOCK-UP status
+                cat ${MONITOR_LOCKED_FILE} | grep "LOCKED"                
+                if [ $? -eq 0 ];then
+                    #KEEP LOCK-UP status
+                    #To let just only one virsh console with one vm guest as backgroup PID			        
+			       
+			        #Check with background PID of virsh console
+			        for vmguest in ${vm_guestnames_array[@]};do
+			            virsh list --state-running | grep ${vmguest}
+			                if [[ $? -eq 0 ]];then
+			                    ps aux | grep "virsh console ${vmguest}"
+                                if [[ $? -ne 0 ]];then
+                                    get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+                                fi
+			                else			                    
+			                    virsh list --inactive | grep ${vmguest}
+			                    if [[ $? -eq 0 ]];then
+			                        echo "${vmguest} is not running now"
+			                        echo "No any output from virsh console ${vmguest}"
+			                    fi                            
+			                fi
+			        done
+			    else
+                    #SETUP UNLOCKED
+			        echo "UNLOCK Monitor for $(basename $0)!"
+			        exit			        
+			    fi
+		    else
+			    echo "Create LOCK-UP for $(basename $0)"
+			    echo "LOCKED" > ${MONITOR_LOCKED_FILE}
+                open_console
+		    fi
+	    done
```

Comment:
It should be OK, but need to pay attention to whether frequent while loop without sleep functions stably. If it turns out to be "prone to have problem", please do not add long time of sleep, 1s-3s is enough.